npm/test/clientfilemetataupload.test.ts

```diff
@@ -194,6 +203,41 @@ test("saveClientFileMetadata fails to upload client file metadata", async () =>
   await expect(
     uploadMetadata.saveClientFileMetadata(fileIds, metadata)
   ).rejects.toStrictEqual(
-    Error("Add client file metadata failed for file 1: error 1,error 2")
+    Error("Add client file metadata failed: error 1,error 2")
   )
 })
+
+test("createMetadataInputBatches generates batches of the defined size", () => {
+  const input1 = generateMockMetadataInput()
+  const input2 = generateMockMetadataInput()
+  const input3 = generateMockMetadataInput()
+  const input4 = generateMockMetadataInput()
+  const input5 = generateMockMetadataInput()
+
```

Comment:
The only thing I'm not sure of in this test is that each of these inputs has the same values, even with the Date.now() which is weird. This means that you can swap out the order of the inputs in the expect statement and the test still passes. I don't know if this matters because the point is that it splits them into arrays of the right size, not that they're in the right order. So if you think the order doesn't matter, then ignore me. If it does, you can make them unique by putting `Math.random().toString(36)` in the fileId and then the ordering matters.

Reply:
Yeah good point. The order isn't particular important, but I think it should be checked as there is quite a lot of iterating through, and its good to keep the ordering consistent, I think. Made the change here to make the inputs unique to test the ordering corectly.